org.eclipse.scanning.sequencer/src/org/eclipse/scanning/sequencer/analysis/SliceDeviceContext.java

```diff
@@ -0,0 +1,90 @@
+package org.eclipse.scanning.sequencer.analysis;
+
+import org.eclipse.january.dataset.IDataset;
+import org.eclipse.january.dataset.ILazyDataset;
+import org.eclipse.scanning.api.points.IPosition;
+import org.eclipse.scanning.api.scan.rank.IScanSlice;
+
+public class SliceDeviceContext {
+
+	private IPosition    location;
+	private IScanSlice   scanSlice;
+	private ILazyDataset data;
+	private IDataset     slice;
+	
+	public SliceDeviceContext() {
+		
+	}
+	public SliceDeviceContext(IPosition loc, IScanSlice rslice, ILazyDataset data, IDataset slice) {
+		this.location    = loc;
+		this.scanSlice = rslice;
+		this.data   = data;
+		this.slice  = slice;
+	}
+	public IPosition getLocation() {
+		return location;
+	}
+	public void setLocation(IPosition loc) {
+		this.location = loc;
+	}
+	public IScanSlice getScanSlice() {
+		return scanSlice;
+	}
+	public void setScanSlice(IScanSlice rslice) {
+		this.scanSlice = rslice;
+	}
+	public ILazyDataset getData() {
+		return data;
+	}
+	public void setData(ILazyDataset data) {
+		this.data = data;
+	}
+	public IDataset getSlice() {
+		return slice;
+	}
+	public void setSlice(IDataset slice) {
+		this.slice = slice;
+	}
+	@Override
+	public int hashCode() {
+		final int prime = 31;
+		int result = 1;
+		result = prime * result + ((data == null) ? 0 : data.hashCode());
+		result = prime * result + ((location == null) ? 0 : location.hashCode());
+		result = prime * result + ((scanSlice == null) ? 0 : scanSlice.hashCode());
+		result = prime * result + ((slice == null) ? 0 : slice.hashCode());
+		return result;
+	}
+	@Override
+	public boolean equals(Object obj) {

```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 19 to the 15 allowed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3776)